src/main/java/jraft/impl/RaftServerContextImpl.java

```diff
@@ -234,6 +236,7 @@ private void doAsCandidate() {
             } else {
                 if (2 * voteCounter > peerMap.size() + 1) {
                     setTerm(this.term + 1);
+                    setLastVoteFor(null);
```

Comment:
may not needed because for each transition call, the first thing it does is to clear every old data from previous state.